libbeat/scripts/aggregate_coverage.py

```diff
@@ -28,9 +28,10 @@ def main(arguments):
     args.outfile.write('mode: atomic\n')
     for m in matches:
         with open(m) as f:
-            for line in f:
-                if not line.startswith('mode:'):
-                    args.outfile.write(line)
+            if os.path.abspath(args.outfile.name) != os.path.abspath(f.name):
```

Comment:
why not compare before opening the file? os.path.abspath(args.outfile.name) != os.path.abspath(m)